microsoft/testsuites/vm_extensions/azsecpack.py

```diff
@@ -0,0 +1,287 @@
+        resource_group_name = node_context.resource_group_name
+        location = node_context.location
+        vm_name = node_context.vm_name
+
+        # Add resource tag for AzSecPack
+        tag = {"azsecpack": "nonprod"}
+        add_tag_for_vm(platform, resource_group_name, vm_name, tag, log)
+
+        # Create an AzSecPackAutoConfigRG resource group with the specific location
+        # For Public cloud, the specific location is "eastus".
+        autoconfig_rg_name = "AzSecPackAutoConfigRG"
+        autoconfig_rg_exists = rm_client.resource_groups.check_existence(
+            autoconfig_rg_name
+        )
+        if not autoconfig_rg_exists:
+            params = {"location": "eastus"}
+            rm_client.resource_groups.create_or_update(autoconfig_rg_name, params)
+            log.info(f"{autoconfig_rg_name} is created successfully")
+        else:
+            log.info(f"{autoconfig_rg_name} is already existed")
+
+        # Create a user assigned managed identity
+        msi_name = f"AzSecPackAutoConfigUA-{location}"
+        msi = msi_client.user_assigned_identities.create_or_update(
+            resource_group_name=autoconfig_rg_name,
+            resource_name=msi_name,
+            parameters={"location": location},
+        )
+        log.info(f"{msi.id} is created successfully")
+
+        # Assign the user assigned managed identity to the VM
+        add_user_assign_identity(platform, resource_group_name, vm_name, msi.id, log)
+
+        # Install agent extension
+        self._install_monitor_agent_extension(node)
+        self._install_security_agent_extension(node)
+        self._install_mdsd(node, log)
+
+        # Check and verify
+        self._check_mdsd_service_status(node, log)
+        self._check_azsec_services_status(node, log)
+        self._check_azsecd_status(node, log)
+        self._check_journalctl_logs(node, log)
+        self._check_mdsd_qos_logs(node, log)
+
+        # Clean up the environment
+        if not autoconfig_rg_exists:
+            rm_client.resource_groups.begin_delete(autoconfig_rg_name)
+            log.info(f"{autoconfig_rg_name} is deleted successfully")
+
+    def _install_mdsd(self, node: Node, log: Logger) -> None:
+        package = "azure-mdsd"
+        posix_os: Posix = cast(Posix, node.os)
+        is_installed = posix_os.package_exists(package)
+        if not is_installed:
+            # Add azurecore repo
+            if isinstance(posix_os, CBLMariner):
+                release = posix_os.information.release
+                curl = node.tools[Curl]
+                curl.fetch(
+                    arg="-o /etc/yum.repos.d/mariner-extras.repo",
+                    execute_arg="",
+                    url="https://raw.githubusercontent.com/microsoft/CBL-Mariner/"
+                    f"{release}/SPECS/mariner-repos/mariner-extras.repo",
+                    sudo=True,
+                )
+            else:
+                posix_os.add_azure_core_repo()
```

Comment:
mdm also uses the similar code, please put the common code into common.py under vm_extensions folder, then rerun them

Reply:
Done